lib/Bindings/Python/circt/dialects/_hw_ops_ext.py

```diff
@@ -189,6 +195,55 @@ def create(self,
                            ip=ip)
 
 
+def _create_output_op(cls_name, output_ports, entry_block, bb_ret):
```

Comment:
The automatic output creating is nice, but all the options supported by bb_ret seem a little convoluted. Shouldn't the api require the basic block to return a BuilderValue, Operation, or Value? One of the "value-like" abstractions?

Reply:
I don't understand... How would we support an array of values? (Which is required for multiple outputs.)

Reply:
Sorry, I was thinking about the multiple outputs case, but I forgot to mention. If I ammend my previous comment to "return a BuilderValue, Operation, or Value, or a tuple of the above" does it make sense? That it what I meant to drive at.